### PR TITLE
Filter in InstructorFeedbackResultsPage not working #2759

### LINKED ISSUE #2759: Filter in InstructorFeedbackResultsPage not working
For "Group by question", the filter works properly. However, for the other 4 view types (Giver > Recipient > Question, Recipient > Giver > Question, Giver > Question > Recipient, Recipient > Question > Giver), the filter does not seem to work as advertised.

### PR DESCRIPTION
**Pull request by rcyq**
Fixes #2759

### CODE REVIEW

File: src/main/webapp/js/instructorFeedbackResults.js [hunk 1]
```diff
@@ -77,25 +77,90 @@ function showHideStats(){
     }
 }
 
-//Search functionality
-
+// Filter functionality
+// Filtering is done by searching the heading text in all panels (section, team name, student name)
+//
+// When the heading text of a panel has found to be matched with the search text, 
+// all nested panels in the panel will be shown and parent panels of the panel will be shown as well
```
Comment (LowWeiLin) [inline src/main/webapp/js/instructorFeedbackResults.js]: When the heading text of a panel is found to match the search text,

File: src/main/webapp/js/instructorFeedbackResults.js [hunk 2]
```diff
@@ -77,25 +77,90 @@ function showHideStats(){
...
+//
+// When at least one of the nested panels in a panel has found to be matched with the search text,
```
Comment (LowWeiLin) [inline src/main/webapp/js/instructorFeedbackResults.js]: When at least one of the nested panels in a panel is found to contain the search text in its title,

File: src/main/webapp/js/instructorFeedbackResults.js [hunk 3]
```diff
@@ -77,25 +77,90 @@ function showHideStats(){
...
         } else {
-           $(elm).hide();
-        } 
+            // current panel text does not match with search text & current panel has child panels
+            // add current panel to pending parent panel stack
+            showStack.push(panel);
+        }
+
+        if(hasChild){
+            // push current panel whhen it has child panels
```
Comment (LowWeiLin) [inline src/main/webapp/js/instructorFeedbackResults.js]: typo

**Comment by LowWeiLin** [comment]
For question view: "Type keywords from the question title" can be "Type keywords from the question to filter" to be more consistent with the other views.

For the other views, I cannot search by question text. Is that expected?
Should also say in the placeholder text that we can search by tutorial group.

**Comment by rcyq** [comment]
@LowWeiLin yes, the filter function can only search panel text. It is expected that instructor cannot filter by question other than "question view"

File: src/main/webapp/jsp/instructorFeedbackResultsTop.jsp [hunk 4]
```diff
@@ -168,7 +168,7 @@
                         </label>
                         <div class="col-sm-10">
                             <div class="input-group">
-                                <input type="text" id="results-search-box" class="form-control" placeholder='<%= data.sortType.equals("question") ? "Type keywords from the question title" : "Type a student/team name to filter results"%>' onchange="updateResultsFilter()">
+                                <input type="text" id="results-search-box" class="form-control" placeholder='<%= data.sortType.equals("question") ? "Type keywords from the question to filter" : "Type student/team name/tutorial group to filter"%>' onchange="updateResultsFilter()">
```
Comment (LowWeiLin) [inline src/main/webapp/jsp/instructorFeedbackResultsTop.jsp]: This should be "section name" instead of "tutorial group"?

**Comment by LowWeiLin** [comment]
Sorry I mentioned wrongly to use the term "tutorial group" in my previous comment. Should change to "section name". After than that, it looks good.

**Pull request merged**